Script/AtomicEditor/hostExtensions/languageExtensions/TypscriptLanguageExtension.ts

```diff
@@ -195,7 +270,151 @@ export default class TypescriptLanguageExtension implements Editor.HostExtension
+            this.isTypescriptProject = true;
+        });
+
+        this.configureTypescriptProject();
+    }
+
+
+    /**
+     * Called when the project is unloaded
+     */
+    projectUnloaded() {
+        // Clean up
+        this.serviceRegistry.uiServices.removePluginMenuItemSource("TypeScript");
+        this.compileOnSaveMenuItem = null;
+        this.isTypescriptProject = false;
+    }
+
+    /*** UIService implementation ***/
+
+    /**
+     * Called when a plugin menu item is clicked
+     * @param  {string} refId
+     * @return {boolean}
+     */
+    menuItemClicked(refId: string): boolean {
+        let [extension, action] = refId.split(".");
+        if (extension == this.name) {
+            switch (action) {
+                case "compileonsave":
+                    let isCompileOnSave = this.serviceRegistry.projectServices.getUserPreference(this.name, "CompileOnSave", false);
+                    // Toggle
+                    isCompileOnSave = !isCompileOnSave;
+                    this.serviceRegistry.projectServices.setUserPreference(this.name, "CompileOnSave", isCompileOnSave);
+                    this.compileOnSaveMenuItem.string = `Compile on Save: ${isCompileOnSave ? "On" : "Off"}`;
+                    return true;
+                case "compileproject":
+                    this.doFullCompile();
+                    return true;
+            }
+        }
+    }
+
+    /**
+     * Handle messages that are submitted via Atomic.Query from within a web view editor.
+     * @param message The message type that was submitted to be used to determine what the data contains if present
+     * @param data any additional data that needs to be submitted with the message
+     */
+    handleWebMessage(messageType: string, data: any) {
+        switch (messageType) {
+            case "TypeScript.DisplayCompileResults":
+                this.displayCompileResults(data);
+                break;
+        }
+    }
+
+    setTsConfigOnWebView(tsConfig: any) {
+        WebView.WebBrowserHost.setGlobalStringProperty("TypeScriptLanguageExtension", "tsConfig", JSON.stringify(tsConfig));
+    }
+
+    /**
+     * Perform a full compile of the TypeScript
+     */
+    doFullCompile() {
+        const editor = this.serviceRegistry.uiServices.getCurrentResourceEditor();
+        if (editor && editor.typeName == "JSResourceEditor" && this.isValidFiletype(editor.fullPath)) {
+            const jsEditor = <Editor.JSResourceEditor>editor;
+            jsEditor.webView.webClient.executeJavaScript(`TypeScript_DoFullCompile();`);
+        } else {
+            this.serviceRegistry.uiServices.showModalError("TypeScript Compilation", "Please open a TypeScript file in the editor before attempting to do a full compile.");
+        }
+
+        // Ideally, we would want to either launch up a background web view, or shell out to node or something and not
+        // need to have an editor open.  Still researching this
```

Comment:
Ok, yeah... also, I have progress on updating to new Chromium, we'll see how this week sorts, still dealing with some stuff here, I think node could be a good option